packages/frontend/types/emberclear/index.d.ts

```diff
@@ -101,14 +102,29 @@ declare global {
     time: string;
   }
 
-  interface ChannelInvitation {
-    invitePublicKey: string;
-    result: MemberResult[];
+  interface StandardVoteChain {
```

Comment:
This is def out of scope, but something to keep in mind (for myself :) )

- Would it be beneficial to have the Members stored in a unified Map of some sort?
   - This would allow the remaining,yes,no arrays to just be the keys of that map.
- Do we  need the full ChannelMember data in each of remaining, yes, and no?
   - is this data duplicated somewhere?
   - is there a "source of truth"?

Reply:
So within a voteChain, the "source of truth" regarding who can vote on that action is the base vote. When someone creates a vote, they put all the members from their context in remaining. When somebody starts an action and sends out their vote, other members would invalidate it if the yes/no/remaining pool of members was not equivalent to their list of members in their most updated channel context.

We don't technically need the full ChannelMember data in each remaining, yes, and no. In reality you only need the public keys.

One issue that we've considered is whether or not an active vote should be invalidated if another passes or if the context of a vote should essentially stay what it was when a vote started. It gets into some murky territory, but either would work. I feel the second way is better from the user perspective because it would be obnoxious to have to keep submitting actions if other passed. That being said, there are a lot more edge cases associated with that. For example, a vote could get into an incompletable state if enough of the voting members are removed from the channel before that vote can pass or fail.